packages/mosaic-tanstack-table-core/src/data-table.ts

```diff
@@ -400,26 +401,46 @@ export class MosaicDataTable<TData extends RowData, TValue = unknown>
 
   override queryResult(table: unknown): this {
     if (isArrowTable(table)) {
+      const arrowResult = table as unknown as ArrowTable;
       let totalRows: number | undefined = undefined;
 
-      const rows = table.toArray() as Array<TData>;
-
-      if (
-        this.options.totalRowsMode === 'window' &&
-        rows.length > 0 &&
-        rows[0] &&
-        typeof rows[0] === 'object' &&
-        this.#sql_total_rows in rows[0]
-      ) {
-        const firstRow = rows[0] as Record<string, any>;
-        totalRows = firstRow[this.#sql_total_rows];
+      // LAZY OPTIMIZATION:
+      // Instead of converting the entire Arrow table to JS objects (which freezes the main thread),
+      // we generate a lightweight array of index objects. The column accessors (wrapped below)
+      // will fetch the actual data lazily from `arrowResult`.
+      const rowIndices = Array.from(
```

Comment:
This does feel a bit weird, since we're anyways creating objects here. This was earlier being done by `table.toArray()` which came from Mosaic's fletchette anyways.

Not sure if there anything you can do about it, but just thought I'd mention it here.